testsuite/classlibrary/TestUGen_initSample.sc

```diff
@@ -0,0 +1,253 @@
+TestUGen_initSample : UnitTest {
+
+	var server;
+
+	*initClass {
+		passVerbosity = UnitTest.brief;
+	}
+
+	setUp {
+		server = Server(this.class.name);
+		server.bootSync;
+	}
+
+	tearDown {
+		server.quit;
+		server.remove;
+	}
+
+	// Async helpers
+
+	awaitSynthOutput { |synthFn, runtimeSamples=1, timeout=1|
+		var cond = CondVar(), synthOutput = nil, runtimeSeconds;
+		// turn samples in blocks if synthFn is control-rate
+		if (synthFn.asSynthDef.children.last.rate !== \audio) {
+			runtimeSamples = runtimeSamples * server.options.blockSize
+		};
+		runtimeSeconds = runtimeSamples / server.sampleRate;
+		// allow extra timeout for longer tests (e.g. testFFT with big memSize)
+		if (runtimeSeconds > timeout) {
+			timeout = runtimeSeconds + timeout
+		};
+		synthFn.loadToFloatArray(runtimeSeconds, server) { |out|
+			synthOutput = out;
+			cond.signalOne();
+		};
+		cond.waitFor(timeout) { synthOutput.notNil };
+		^synthOutput;
+	}
+
+	assertSynthOutput { |synthFn, targetOutput, timeout=1|
+		var nSamples = targetOutput.size;
+		// var synthFnCode = synthFn.def.sourceCode.replaceRegexp("[ \n\r\t]","");
+		var synthFnCode = synthFn.def.sourceCode;
+		var out = this.awaitSynthOutput(synthFn, nSamples);
+		if (out.isNil) {
+			this.assert(false, "% timed out (timeout: %)".format(synthFnCode, timeout))
+		} {
+
+			this.assertArrayFloatEquals(out.keep(nSamples), targetOutput,
+				"% should produce expected output".format(synthFnCode)
+			);
+		};
+	}
+
+	assertInitEqualsFirstSample { |synthFn|
+		var out = this.awaitSynthOutput({
+			var in = synthFn.value;
+			var initSample = if (in.rate == \audio) {
+				InitializationSample.ar(in)
+			} {
+				InitializationSample.ir(in)
+			};
+			[initSample, in]
+		}, 1);
+		var fnCode = synthFn.def.sourceCode.replaceRegexp("[ \n\r\t]","");
+		this.assertFloatEquals(out[0], out[1],
+			"% init sample and first sample should be equal".format(fnCode));
+	}
+
+	// Tests
+
+	test_InitializationSample {
+		this.assertSynthOutput(
+			{ InitializationSample.ar(1) }, [1.0, 1.0],
+			"InitializationSample"
+		);
+		this.assertSynthOutput(
+			{ InitializationSample.ar(0) }, [0.0, 0.0],
+			"InitializationSample"
+		);
+		this.assertInitEqualsFirstSample { InitializationSample.ar(1) };
+		this.assertInitEqualsFirstSample { InitializationSample.ar(0) }
+	}
+
+	test_TriggerUGens {
+		var basicTest = { |ugenClass|
+			var tests = [];
+			if(ugenClass.respondsTo(\ar)) {
+				tests = tests ++ [
+					"{%.ar(DC.ar(0))}",
+					"{%.ar(DC.ar(1))}",
```

Comment:
It would be good to use less common "default" values for the input signal. E.g. a problematic UGen might have it's init sample hard-coded to `0.0`, so the test signal `DC.ar(0)` wouldn't catch the bug.

Reply:
the intention was to test triggers, but I see your point and we could use -1234 and 1234 instead